### DISCUSSION TITLE
Monitoring Dashboard for vLLM

### DISCUSSION THREAD

**Discussion by jasonacox** [vllm-project/vllm#4700] [General]
## vLLM Dashboard

What are you using to monitor your vLLM installation? 

I wanted a simple dashboard to help trend vLLM and GPU metrics and performance. The Prometheus metrics that vLLM provides gives some good insight into the health and use of the system with data like tokens per second, running requests, waiting requests and more. What indicators do you find most useful?

## Grafana for vLLM and GPU Metrics

I put together a Grafana dashbaord using InfluxDB as the datastore, Telegraf to pull the metrics from vLLM and a custom script to grab nvidia-smi data. I uploaded the details here and would be happy to create an example PR if there is interest: https://github.com/jasonacox/TinyLLM/tree/main/monitoring

<img width="1214" alt="image" src="https://github.com/vllm-project/vllm/assets/836718/f03a9aa3-cac1-43a6-b51f-91b62b2b869d">

![image](https://github.com/jasonacox/TinyLLM/assets/836718/a1389cb0-c3d1-46ec-bec1-1ff3ac412507)

**Comment by simon-mo** [comment]
https://github.com/vllm-project/vllm/tree/main/examples/production_monitoring

Here's an exampel dashboard put together by @robertgshaw2-neuralmagic

**Reply by jasonacox** [reply]
Thanks @simon-mo ! I'm going to use that. 

@robertgshaw2-neuralmagic Could you share your [grafana.json ](https://raw.githubusercontent.com/vllm-project/vllm/main/examples/production_monitoring/grafana.json)dashboard file using the "Export for sharing externally" toggle? The one that is there seems to be using hardcoded datasources.

<img width="628" alt="image" src="https://github.com/vllm-project/vllm/assets/836718/ae52b3a7-70bb-4f11-8ca1-7424cf253a06">

**Reply by robertgshaw2-redhat** [reply]
@jasonacox https://github.com/vllm-project/vllm/pull/4711/files should fix.

thanks for pointing this out, not sure how this got into this state

**Reply by frittentheke** [reply]
@robertgshaw2-neuralmagic I am using the Dashboard via automatic import from K8s ConfigMaps 
(something along the lines of https://artifacthub.io/packages/helm/kiwigrid/grafana-dashboards)

I changed the datasource to come from a template var in: https://github.com/vllm-project/vllm/pull/6136
to account for these kind of use cases - PTAL.
